pkg/config/config.go

```diff
@@ -37,37 +37,37 @@ var (
 )
 
 type ClusterConfig struct {
-	URL string `json:"url"`
-
-	ClusterCIDR          string `json:"clusterCIDR"`
-	ServiceCIDR          string `json:"serviceCIDR"`
-	ServiceNodePortRange string `json:"serviceNodePortRange"`
-	DNS                  string `json:"dns"`
-	Domain               string `json:"domain"`
-	MTU                  string `json:"mtu"`
+	// URL string `json:"url"`
+
+	// ClusterCIDR          string `json:"clusterCIDR"`
+	// ServiceCIDR          string `json:"serviceCIDR"`
+	// ServiceNodePortRange string `json:"serviceNodePortRange"`
+	// DNS string `json:"dns"`
+	// Domain               string `json:"domain"`
+	// MTU string `json:"mtu"`
 }
 
 type DebugConfig struct {
-	Pprof bool `json:"pprof"`
+	// Pprof bool `json:"pprof"`
 }
 
 type MicroshiftConfig struct {
-	ConfigFile string `json:"configFile"`
-	DataDir    string `json:"dataDir"`
-
-	AuditLogDir string `json:"auditLogDir"`
-	LogVLevel   int    `json:"logVLevel"`
-
-	Roles []string `json:"roles"`
-
-	NodeName string `json:"nodeName"`
-	NodeIP   string `json:"nodeIP"`
-
-	Cluster ClusterConfig `json:"cluster"`
-
-	Manifests []string `json:"manifests"`
-
-	Debug DebugConfig `json:"debug"`
+	// ConfigFile string `json:"configFile"`
+	// DataDir    string `json:"dataDir"`
+	//
+	// AuditLogDir string `json:"auditLogDir"`
+	// LogVLevel   int    `json:"logVLevel"`
+	//
+	// Roles []string `json:"roles"`
+	//
+	// NodeName string `json:"nodeName"`
```

Comment:
We'll eventually need this override as host names cannot be guaranteed to be unique or stable.

Reply:
It could be selected arbitrarily (even randomly) and persisted to address uniqueness and stability. Is uniqueness an important consideration for MicroShift? My understanding is that there would never be more than one node, so even a fixed node name would be fine. What happens if a user reconfigures the node name after first start?

Reply:
We're starting with single node, because that's where 80% of the use cases are, but expect to need 3-node (HA) eventually. I guess we can cross that bridge when we get there, though.

Right now, the use cases I believe we need to address the following scenarios:
a) User has resource manifests selecting on node (name) on OCP that shall run unmodified on MicroShift, too. Could be handled by defaulting node name == host name (current behaviour).
b) Host changes name during runtime, either as result of DHCP lease renewal or the device owner / device agent deliberately changing name from the image defaults for better host identification after MicroShift booted for the first time.

Reply:
Uniqueness will be important for fleet management, but do we have a separate cluster ID for that? Can we have MicroShift set the node name to a hardware device ID of some sort instead of asking users to configure it?

Reply:
If we agree we can use a hardware ID, we'll want to leave the field in place for now and file a ticket to do the work to change it later.